Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
+                                if (currMismatch > 0 || entityMatch.Count == 0)
                                 {
-                                    entities.Add(
-                                        new Entity
-                                        {
-                                            Name = match.Value,
-                                            MatchIndex = match.Index
-                                        });
+                                    if (!entityMatch.Where(x => (x.Index >= currTextChar - (wordCharArray.Count + leniency)) &&
+                                    (x.Index <= currTextChar - wordCharArray.Count)).Any())
+                                    {
+                                        valuesList.Add(wordFound.ToString());
+                                        offsetList.Add(currTextChar - wordFound.ToString().Length);
+                                        confidenceList.Add(currMismatch + (wordCharArray.Count - currWordChar) / leniency);
+                                        if (minVal < 0 || wordFound.ToString().Length < minVal)
+                                            minVal = wordFound.ToString().Length;
+                                        addWord = true;
+                                    }
+                                    currWordChar = KMPTable[currWordChar];
+                                    wordFound.GetStringBuilder().Clear();
+                                    currMismatch = 0;
                                 }
-                                entitiesFound.Add(word);
+                                else
+                                {
+                                    currWordChar = KMPTable[currWordChar];
+                                    if (currWordChar < 0)
+                                    {
+                                        currWordChar++;
+                                        currTextChar++;
+                                    }
+                                    wordFound.GetStringBuilder().Clear();
+                                    currMismatch = 0;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            string accents = @"[\p{Mn}\p{Sk}\p{Mc}]";
+                            bool wordHalfMark = Regex.Match(wordCharArray[currWordChar].ToString(), accents).Success;
+                            bool textHalfMark = Regex.Match(textCharArray[currTextChar].ToString(), accents).Success;
+
+                            // fuzzy situation?
+                            // accent case adds 0.5
+                            if ((wordHalfMark && textHalfMark) ||
+                                String.Compare(wordCharArray[currWordChar].ToString(), textCharArray[currTextChar].ToString(),
+                                CultureInfo.CurrentCulture, CompareOptions.IgnoreCase | CompareOptions.IgnoreNonSpace) == 0)
```

Comment:
Do you really need to do both the regex tests and the culture-aware string comparison?

Looks like the regex case is too wide, and will catch any pair of non-matching accented characters.

Are you sure the current culture should be used here? We know what language the text is in, so we should be able to infer a culture from it and use that.

Reply:
The issue currently is I do not ask for the language and the current culture is always english (and hence unnecessary will delete). Both are needed as in some instances the accent is considered nonspacing but still a different unicode value.

Reply:
Also, as seen w. the test, it does not equate for instance a & e

Reply:
You can't assume current culture is English. Since the skill deals with aspects of language that are not found in English, it may be reasonable actually to add language as an optional parameter of the skill. It's not a big constraint as language is already available from most skillsets.

On both being needed, reversely, are there cases that the string comparison would not find that the regexp would?

Reply:
Ah, I'm sorry. I meant that whenever I ran my tests with different languages, the culture remained english since its based on my UI (I think?). I am now just using the char class checks as that's all im really using the regexp for